### PR TITLE
CommCareMultimedia attachments -> blobs

### PR DESCRIPTION
**Pull request by millerdev**
This has been on staging for a while now with no issues. Review commits individually.

@dannyroberts @snopoke cc @benrudolph

### CODE REVIEW

File: corehq/apps/hqmedia/models.py [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
-from PIL import Image
 from datetime import datetime
-import hashlib
+from StringIO import StringIO
+
+import magic
 from couchdbkit.exceptions import ResourceConflict
 from dimagi.ext.couchdbkit import *
-from corehq.apps.app_manager.exceptions import XFormException
+from dimagi.utils.couch.database import get_db, get_safe_read_kwargs, iter_docs
```
Comment (dimagimon) [inline corehq/apps/hqmedia/models.py]: F401 'get_db' imported but unused

**Comment by benrudolph** [comment]
👍 test failure looks unrelated:
```
======================================================================
ERROR: test_get_call_center_domains (corehq.apps.callcenter.tests.test_utils.CallCenterDomainTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/apps/callcenter/tests/test_utils.py", line 347, in test_get_call_center_domains
    _create_domain('cc-dom1', True, True, 'flw')
  File "/mnt/corehq/apps/callcenter/tests/test_utils.py", line 376, in _create_domain
    doc=domain.to_json(),
  File "/mnt/corehq/elastic.py", line 54, in send_to_elasticsearch
    doc_exists = doc_exists_in_es(index, doc_id)
  File "/mnt/corehq/elastic.py", line 44, in doc_exists_in_es
    return get_es_new().exists(es_meta.index, es_meta.type, doc_id)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/client/utils.py", line 69, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/client/__init__.py", line 290, in exists
    id), params=params)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/transport.py", line 307, in perform_request
    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/connection/http_urllib3.py", line 93, in perform_request
    self._raise_error(response.status, raw_data)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/connection/base.py", line 105, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
TransportError: TransportError(503, u'')
```

**Comment by snopoke** [comment]
Debugging that now

File: corehq/apps/hqmedia/models.py [hunk 2]
```diff
@@ -170,9 +175,9 @@ def add_domain(self, domain, owner=None, **kwargs):
 
     def get_display_file(self, return_type=True):
         if self.attachment_id:
-            data = self.fetch_attachment(self.attachment_id, True).read()
+            data = self.fetch_attachment(self.attachment_id, stream=True).read()
```
Comment (snopoke) [inline corehq/apps/hqmedia/models.py]: nice

**Comment by snopoke** [comment]
:+1: on :white_check_mark:

**Comment by snopoke** [comment]
:boot: :construction:

**Comment by millerdev** [comment]
same failure

**Comment by millerdev** [comment]
Unfortunately the tests for this PR were not run:

#31426.2 passed --divided-we-run=08
`corehq.blobs.tests.test_migrate.TestMultimediaMigrations ... SKIP: divided-we-run: c not in range 08`

#31426.3 failed --divided-we-run=9f
no mention of `TestMultimediaMigrations` in tests that were run

**Comment by dannyroberts** [comment]
same failure here by the way: https://github.com/dimagi/commcare-hq/pull/11726#issuecomment-221365928. From @snopoke's comments, it looks like he must have fixed it on master? Maybe all this needs is to merge master?

**Comment by snopoke** [comment]
Hopefully fixed here: https://github.com/dimagi/commcare-hq/pull/11720

**Comment by millerdev** [comment]
✅

**Pull request merged**